osu.Game/Scoring/Legacy/ScoreInfoExtensions.cs

```diff
@@ -12,14 +12,21 @@ namespace osu.Game.Scoring.Legacy
 {
     public static class ScoreInfoExtensions
     {
-        public static long GetDisplayScore(this ScoreProcessor scoreProcessor, ScoringMode mode)
-            => getDisplayScore(scoreProcessor.Ruleset.RulesetInfo.OnlineID, scoreProcessor.TotalScore.Value, mode, scoreProcessor.MaximumStatistics);
+        public static long GetDisplayScore(this ScoreProcessor scoreProcessor, ScoringMode mode, bool noScoreMultiplier = false)
+            => getDisplayScore(scoreProcessor.Ruleset.RulesetInfo.OnlineID, scoreProcessor.TotalScore.Value, scoreProcessor.ScoreMultiplier, noScoreMultiplier, mode, scoreProcessor.MaximumStatistics);
 
-        public static long GetDisplayScore(this ScoreInfo scoreInfo, ScoringMode mode)
-            => getDisplayScore(scoreInfo.Ruleset.OnlineID, scoreInfo.TotalScore, mode, scoreInfo.MaximumStatistics);
+        public static long GetDisplayScore(this ScoreInfo scoreInfo, ScoringMode mode, bool noScoreMultiplier = false)
+            => getDisplayScore(scoreInfo.Ruleset.OnlineID, scoreInfo.TotalScore, scoreInfo.ScoreMultiplier, noScoreMultiplier, mode, scoreInfo.MaximumStatistics);
 
-        private static long getDisplayScore(int rulesetId, long score, ScoringMode mode, IReadOnlyDictionary<HitResult, int> maximumStatistics)
+        private static long getDisplayScore(int rulesetId, long score, double scoreMultiplier, bool noScoreMultiplier, ScoringMode mode, IReadOnlyDictionary<HitResult, int> maximumStatistics)
         {
+            if (noScoreMultiplier)
+            {
+                // This might result in a minor inaccuracy of score (+-1 point) because the passed in score is already rounded.
+                // This shouldn't be much of a problem because noScoreMultiplier is only a visual effect meant for fun.
+                score = (long)Math.Round(score / scoreMultiplier);
+            }
```

Comment:
Having to pass `scoreMultiplier` and `noScoreMultiplier` and having to _undo_ an already-applied multiplier feels completely backwards. Much less so that this comment states that the undo _cannot be done accurately_ due to rounding applied.

Reply:
I don't think having a minor inaccuracy is problematic. This feature implements only a visual difference in scores. Having this feature coded alongside `ScoringMode` as final step in calculating score that is displayed makes sense to me. Classic scoring is also just a fake front for users anyway, since internally calculations are done the same way as standardised scoring.

I made `ScoreProcessor` calculate proper score, but `ScoreInfo` still requires score multiplier to be unapplied.

Reply:
> Do you think it be better to ditch `GetDisplayScore`?

I don't know how to answer this question because I don't know why the implementation has swung in this entire different direction. The initial version wherein the application of the multipliers (or lack thereof) was coordinated by `ScoreProcessor` and `MultiplayerPlayer` was miles better because it did not weirdly couple components like what this new variant is doing. Is there a constraint that makes that old implementation unviable that I am missing here?

Reply:
I forgot what the reason was, so I tried to work on previous implementation again [here](https://github.com/ppy/osu/compare/master...timiimit:osu:multi-score-multiplier-option-rework?expand=1). Actually, I probably didn't think enough about what continuing with old implementation would require. ScoreV2 update made all code unfamiliar and it seemed too hard to continue so I switched up my approach. In reality I only needed to add a couple of lines...

The actual reason for new implementation was that I couldn't figure out a way to recalculate score with `ScoreProcessor` anymore. In linked branch I still can't recalculate when score is retrieved from server which is done on the results screen. Before ScoreV2 update, `ScoreProcessor.CalculateScore()` could return score but this method is now gone. This makes implementation still suffer from unappyling score multiplier in `PlaylistResultsScreen`.

Let me know if I should continue with linked branch.